.pipelines/.vsts-vhd-builder-windows.yaml

```diff
@@ -16,87 +17,89 @@ trigger: none
-          -w /go/src/github.com/Azure/AgentBaker \
-          -e CLIENT_ID=${CLIENT_ID} \
-          -e CLIENT_SECRET="$(CLIENT_SECRET)" \
-          -e TENANT_ID=${TENANT_ID} \
-          -e CLASSIC_BLOB=${STORAGE_ACCT_BLOB_URL} \
-          -e CLASSIC_SAS_TOKEN="$(STORAGE_ACCT_SAS_TOKEN)" \
-          -e OS_DISK_SAS=${OS_DISK_SAS} \
-          -e VHD_NAME=${VHD_NAME} \
-          ${DEIS_GO_DEV_IMAGE} make -f packer.mk az-copy
-        displayName: Copying resource to Classic Storage Account
-        condition: eq(variables.COPY_VHD, 'True')
-      - script: |
-          OS_DISK_SAS="$(cat packer-output | grep "OSDiskUriReadOnlySas:" | cut -d " " -f 2)" && \
-          VHD_NAME="$(echo $OS_DISK_SAS | cut -d "/" -f 8 | cut -d "?" -f 1)" && \
-          docker run --rm \
-          -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
-          -w /go/src/github.com/Azure/AgentBaker \
-          -e CLIENT_ID=${CLIENT_ID} \
-          -e CLIENT_SECRET="$(CLIENT_SECRET)" \
-          -e TENANT_ID=${TENANT_ID} \
-          -e CLASSIC_SA_CONNECTION_STRING="$(CLASSIC_SA_CONNECTION_STRING)" \
-          -e STORAGE_ACCT_BLOB_URL=${STORAGE_ACCT_BLOB_URL} \
-          -e VHD_NAME=${VHD_NAME} \
-          -e OS_NAME="Windows" \
-          -e SKU_NAME="2019-Datacenter-Core-smalldisk" \
-          -e OFFER_NAME="Windows" \
-          ${DEIS_GO_DEV_IMAGE} make -f packer.mk generate-sas
-        displayName: Generating publishing info for Windows VHD
-        condition: eq(variables.COPY_VHD, 'True')
-      - task: PublishPipelineArtifact@1
-        inputs:
-          artifact: 'publishing-info'
-          path: 'vhd-publishing-info.json'
-        condition: eq(variables.COPY_VHD, 'True')
-      - script: |
-          SA_NAME="$(cat packer-output | grep "storage name:" | cut -d " " -f 3)" && \
-          docker run --rm \
-          -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
-          -w /go/src/github.com/Azure/AgentBaker \
-          -e CLIENT_ID=${CLIENT_ID} \
-          -e CLIENT_SECRET="$(CLIENT_SECRET)" \
-          -e TENANT_ID=${TENANT_ID} \
-          -e SA_NAME=${SA_NAME} \
-          -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
-          ${DEIS_GO_DEV_IMAGE} make -f packer.mk delete-sa
-        displayName: cleanup - delete temp storage account
-        condition: always()
-      - script: |
-          sudo chown -R $USER:$USER .
-        displayName: cleanup - chown all files in work directory
-        condition: always()
\ No newline at end of file
+- job: build_vhd_windows
+  timeoutInMinutes: 150
+  strategy:
+    maxParallel: 0
+  pool:
+    name: $(BUILD_POOL)
+  steps:
+  - script: |
+      docker run --rm \
+      -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
+      -w /go/src/github.com/Azure/AgentBaker \
+      -e CLIENT_ID=${CLIENT_ID} \
+      -e CLIENT_SECRET="$(CLIENT_SECRET)" \
+      -e TENANT_ID=${TENANT_ID} \
+      -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
```

Comment:
Just curious: why do we need to add this line `SUBSCRIPTION_ID` but aks-engine does not need?

Reply:
This is required by [az-login](https://github.com/Azure/AgentBaker/blob/master/packer.mk#L21) target in agentbaker, which is not included aks-engine

Reply:
Why does aks-engine not need az-login?

Reply:
no idea, but this is introduced in [SIG mode: support generating SIG images](https://github.com/Azure/AgentBaker/pull/17) by , I guess he met error creating sig when not specifying subscription_id error

Reply:
SIG image needs to be stored in a subscription. AKS_E doesnt have SIG